### PR TITLE
Adds the Deep Fryer.

### PR DESCRIPTION
**Pull request by Iamgoofball**
Demo video:
https://www.dropbox.com/s/4h9sttkjglciesp/goof_ruined_it_again.mp4?dl=0

Ports the Deep Fryer from NTstation 13.

You can stick in any object, fry it, and eat it. It has 1 nutrient if it is not a reagent container, and if it is a reagent container, it will be the contents of that food item + 1 nutrient.

When you walk into the kitchen nowadays, you don't see chefs experimenting and playing with shit.

You see chefs cranking out the quickest meal they can so the assistants dont murder them because the chef needed to be "balanced". Trying to balance a gimmick job that wasn't supposed to be important in the first place causes problems. It's simply not fun to play chef anymore.

This is an attempt to bring back the "fun times make whatever you want" kitchen. I don't care about the "balance" of the kitchen, because quite frankly it's fucking food it should be fun not balanced.

If something is stuffed in the deep fryer, you can pull it out if you get to it before the fryer finishes.

Objects are qdelled when the frying process finishes **if it's a food item**, if it is not a food item the item will be contained within the deep fried holder, so that if some clever individual stuffs the Nuke Disk in there, it'll be returned upon eating the deep fried nuke disk.

It's not on the map at the moment to avoid merge conflicts, but R&D can build it for the time being.

### CODE REVIEW

**Comment by Incoming5643** [comment]
Liches' Bane

**Comment by PKPenguin321** [comment]
FUCK YEAHHHHH
:+1: x 10,000,000
Can it deep fry people too? I'll :+1: it regardless, but I'm just curious.

**Comment by Jalleo** [comment]
Did you port the blacklist too that nuke disk issue and special items issue.

**Comment by Xhuis** [comment]
WOOP WOOP :+1:

**Comment by bgobandit** [comment]
no

no no no no no 

no

NO

**Comment by AlexCrimson** [comment]
Oh boy time for this bullshit again. DAE fry toolboxes?! So funneh!

**Comment by Niknakflak** [comment]
This was absolute cancer on NT.  Everyone deep fried everything ever.  TONS of bugs.  TONS of adminhelps where people deep fried important shit.  TONS of instances were the important deep fried shit causes problems because the items would be eaten and gone forever.  TONS of retarded "deep fried deep fried deep fried deep fried deep fried wrench".  TONS of autism which made chefs never cook anything EXCEPT deep fried shit.
No
:-1: No no no no no no no no no no no
:-1: :-1: :-1: :-1: :-1: :-1: 
Not every department needs goofball's autistic touch

This ruined cooking on NT.  You ruined cooking on NT.

No

**Comment by Iamgoofball** [comment]
Maybe the fact that chefs made a lot of deep fried shit shows that they enjoyed the feature?

**Comment by AlexCrimson** [comment]
Remember when the majority of the server voted to revert goofchem? Fun times.

**Comment by Iamgoofball** [comment]
Remember when that's literally irrelevant? Fun times.

**Comment by daman997** [comment]
do it faggot :+1:

**Comment by Supermichael777** [comment]
deep fried cigars

**Comment by Iamgoofball** [comment]
>conflicts
literally what

**Comment by bgobandit** [comment]
the codebase itself does not want this to get in

**Comment by KorPhaeron** [comment]
revert goofchem

**Comment by Durkel** [comment]
Deep fried Deep fried Deep fried Deep fried Deep fried Deep fried captain's id.

**Comment by Lumipharon** [comment]
Yes lets go back to infinite deepfried paper, because that was so funny and interesting last time right?

**Comment by RemieRichards** [comment]
## It's a single level of deep frying

**Comment by Niknakflak** [comment]
Still bad. :-1:

**Comment by Iamgoofball** [comment]
Seeing as people were deep frying paper I think they were enjoying themselves.

**Comment by Incoming5643** [comment]
it wouldn't so much an issue if the item was given back as trash after eating the deep fried item.

**Comment by Iamgoofball** [comment]
http://strawpoll.me/6396518

**Comment by KazeEspada** [comment]
<.< >.> What if non-food items stayed inside and caused shit tons toxins damage until removed via surgery.

**Comment by AlexanderUlanH** [comment]
It'll be only marginally harder to get to the deep fryer than the grinder.  There's nothing wrong with this except that it will make the kitchen sillier, which I don't think most people mind.

**Comment by Fox-McCloud** [comment]
The deep-fried grilled deep-fried gut buster icing cake cake cake cake cake deepfriend sandwhich icing cake cake cake cake cake.

File: code/game/machinery/constructable_frame.dm [hunk 1]
```diff
@@ -434,6 +434,14 @@ to destroy them and players will be able to make replacements.
 							/obj/item/weapon/stock_parts/matter_bin = 1,
 							/obj/item/weapon/stock_parts/manipulator = 1)
 
+/obj/item/weapon/circuitboard/deep_fryer
+	name = "circuit board (Deep Fryer)"
+	build_path = /obj/machinery/deepfryer
+	board_type = "machine"
+	origin_tech = "programming=1"
+	req_components = list(
+							/obj/item/weapon/stock_parts/manipulator = 1)
```
Comment (Razharas) [inline code/game/machinery/constructable_frame.dm]: >Frying with a manipulator
Make it laser at least

**Comment by PKPenguin321** [comment]
Don't give up goof I believe in you

**Comment by Iamgoofball** [comment]
fixing conflicts now

**Comment by KazeEspada** [comment]
Test ABSTRACT and NODROP items, and Holodeck items. Also, Grabs could do something with their affecting, you can always add one of those.

Weird items like the Ninja Sword and mob holders(PAI, Drone Shell) and test those as well.

**Comment by Iamgoofball** [comment]
>i forgot the nodrop checks

Thanks for reminding me about ABSTRACT, forgot that shit existed

**Comment by KazeEspada** [comment]
Also might wanna do a size check, backpacks and fireaxes are a bit big to fit in a fryer.

**Comment by Iamgoofball** [comment]
No.

**Comment by 9600bauds** [comment]
In /vg/ we just have non-snack frying (e.g. toolboxes) disabled by default, can be enabled by a bored admin. It doesn't seem to cause any problems, and people do seem to enjoy it.

**Comment by 9600bauds** [comment]
Also, when sillyfrying is enabled, once again nobody seems to complain and most players find it entertaining, if not simply OK. The potential for abuse is there, but since you need to pray to have it enabled, only chefs with a good-inrentioned gimmick get this power.

**Comment by KazeEspada** [comment]
Deep Fried Cyborg Frame

**Comment by Shadowlight213** [comment]
Deep fried deep fryer when?

**Comment by Dorsisdwarf** [comment]
24-10 on eating deep-fried items literally just being scraping off the oil with your teeth

**Comment by kevinz000** [comment]
Deep fried reactive teleportation armor.

**Comment by Bawhoppen** [comment]
deep fried anomaly core

**Comment by Supermichael777** [comment]
can you grab-slam someone to dunk their head into the frying bucket?

**Comment by WkSavage** [comment]
WHY ARE WE NOT FUNDING THIS?

**Comment by Bawhoppen** [comment]
its actually kind of stupid and gimmicky and should probably not be ported is why

**Comment by 9600bauds** [comment]
It's stupid to you. Is it stupid to other people?

**Comment by Bawhoppen** [comment]
yes?

**Comment by PKPenguin321** [comment]
bawhoppen is stupid to me tbh

**Comment by Bawhoppen** [comment]
,haha aha haha hahahhaahhhhhahah

**Comment by PKPenguin321** [comment]
C'mon goof bring this one home I believe in you
Never give up no matter what they say

**Comment by Snerdly** [comment]
http://i.imgur.com/6nMWTyn.png

**Comment by Fox-McCloud** [comment]
And that---above, is exactly why @Niknakflak and myself consider the deep fryer to be retarded as hell.

It's amusing when it's somewhat limited, but when you can chain it infinitely, it's just dumb.

**Comment by 9600bauds** [comment]
That screenshot is not from this PR and you can't chain deep-frying in this implementation.

**Comment by Iamgoofball** [comment]
That's not even my fryer

**THAT'S THE GOON FRYER**

**Comment by Zth--** [comment]
Damn I had so much fun deep frying stuff in vg. Do it for the keks, then restrict it to only food when it gets old

**Comment by Niknakflak** [comment]
For once, I simply agree with baw.  Stupid abd gimmicky

**Comment by Bawhoppen** [comment]
lets compromise on this and make the deep fryer admin only

**Comment by Iamgoofball** [comment]
Do either of you even play chef?

**Comment by Bawhoppen** [comment]
yes?

**Comment by PKPenguin321** [comment]
liar

**Comment by Bawhoppen** [comment]
cyka

**Comment by 9600bauds** [comment]
Like I said, just limit the frying of non-edible items to adminbus only. That's how it works in /vg/ and it's a welcome addition.

**Comment by tkdrg** [comment]
yes sure let's have a food that doesn't actually feed and can only be used by memester for memes also it will get old in minutes

**Comment by Iamgoofball** [comment]
which is why we give it nutrients but then people whine "m-m-much precious kitchen METAGAME AND BALANCE you see the WHOLE WORLD OF SS13 DEPENDS ON THE KITCHEN"

**Comment by tkdrg** [comment]
I would still be against this even if it could feed people honestly
It's still a gimmick that can only do a single thing and doesn't bring anything interesting to the table apart from frying memes

**Comment by Iamgoofball** [comment]
and? people like frying memes. it makes them happy.

**Comment by PKPenguin321** [comment]
its true

**Comment by Lumipharon** [comment]
Food is literally pointless now.
Someone nerfed hunger to the point that you never need to eat in 99% of rounds, so I don't care any more, nothing gets in the way of the validsalad.

**Comment by rev-benji** [comment]
No double frying no problem. how is anyone against this?  The whole point of this is to make chef an enjoyable role, its one of the more dull jobs on the station.  Also buff hunger pls

**Comment by Lumipharon** [comment]
Because
>When you walk into the kitchen nowadays, you don't see chefs experimenting and playing with shit.
You see chefs cranking out the quickest meal they can so the assistants dont murder them because the chef needed to be "balanced".

When in reality this pr will result in deep fried paper/metal rods (if hunger gets unfucked), and will regardless of hunger being shit, will result in deepfrying important items like fucking ID's an guns and eating them to piss people off.

**Comment by PKPenguin321** [comment]
>Objects are qdelled when the frying process finishes if it's a food item, if it is not a food item the item will be contained within the deep fried holder, so that if some clever individual stuffs the Nuke Disk in there, it'll be returned upon eating the deep fried nuke disk.

read the parent pr before you go full autismo lumi

**Comment by Bawhoppen** [comment]
this PR probably needs a balance tag

**Comment by PKPenguin321** [comment]
ah yes the delicate chef food meta. whatever would the crew do if each burger has 1 more nutrient? surely the whole station would collapse

**Comment by 9600bauds** [comment]
on the contrary, it would burst

**Comment by Lumipharon** [comment]
That's even worse though. It's literally just eating the grease off a piece of paper then putting it back.
It's the equivilent of having a button on the wall that refills your hunger instead of a kitchen.

The good thing about the deep frier tbh is the future possibility of dunking people's heads into them as a traitor chef.

**Comment by PKPenguin321** [comment]
If you're gonna sit around and deep fry the same piece of paper 20 times maybe you deserve to have your hunger refilled at that point

**Comment by 9600bauds** [comment]
yyeah it'd be easier to cook something

you could also just grow infinite nutriment next door

Also this wouldn't even be a problem if you just disallow non-edible frying

**Comment by Bawhoppen** [comment]
it's still balance even if you think it would be a good change

**Comment by 9600bauds** [comment]
i berieve in u

goofball

**Comment by Iamgoofball** [comment]
Thanks for the new background

**Comment by PKPenguin321** [comment]
Goooofbaaaaaaall

**Comment by tkdrg** [comment]
I'm not merging this for the reasons I've already explained in the comments
and clearly this isn't getting merged by anybody else either

**Pull request closed without merging**